packages/story-editor/src/components/library/panes/shopping/productList.js

```diff
@@ -0,0 +1,224 @@
+import {
+  Text,
+  THEME_CONSTANTS,
+  Icons,
+  Button,
+  BUTTON_TYPES,
+  BUTTON_SIZES,
+  BUTTON_VARIANTS,
+} from '@googleforcreators/design-system';
+import styled, { css } from 'styled-components';
+import { __, sprintf } from '@googleforcreators/i18n';
+
+/**
+ * Internal dependencies
+ */
+import useProductNavigation from './useProductNavigation';
+import { formatPrice, isOnPage } from './util';
+
+const StyledListItem = styled.div`
+  display: flex;
+  width: 100%;
+  padding: 5px 0;
+  align-items: center;
+  border-bottom: 2px solid ${({ theme }) => theme.colors.divider.tertiary};
+  transition: background-color ease-in-out 300ms;
+  &:focus-within,
+  &:hover,
+  &:focus {
+    background-color: ${({ theme }) => theme.colors.interactiveBg.disable};
+  }
+`;
+
+const baseIcon = css`
+  width: 16px;
+  height: 16px;
+  border-radius: 100%;
+  color: ${({ theme }) => theme.colors.interactiveBg.disable};
+`;
+
+const StyledActionsContainer = styled.div`
+  margin-right: 10px;
+
+  .check {
+    ${baseIcon}
+    background-color: ${({ theme }) => theme.colors.fg.positive};
+  }
+
+  .remove {
+    ${baseIcon}
+    background-color: ${({ theme }) => theme.colors.fg.negative};
+    display: none;
+  }
+
+  button {
+    margin-right: 10px;
+    &:focus,
+    &:hover {
+      .remove {
+        display: block;
+      }
+      .check {
+        display: none;
+      }
+    }
+  }
+`;
+
+const imgPlaceholder = css`
+  display: block;
+  margin-right: 10px;
+  border-radius: 4px;
+  background-color: ${({ theme }) => theme.colors.interactiveBg.disable};
+  width: 41px;
+  height: 41px;
+`;
+
+const StyledImgPlaceHolder = styled.div`
+  ${imgPlaceholder}
+`;
+
+const StyledImage = styled.img`
+  ${imgPlaceholder}
+`;
+
+const StyledDescription = styled.div`
+  padding-left: 12px;
+`;
+
+const StyledTitle = styled(Text).attrs({
+  size: THEME_CONSTANTS.TYPOGRAPHY.PRESET_SIZES.MEDIUM,
+})`
+  color: ${({ theme }) => theme.colors.fg.primary};
+`;
+
+const StyledPrice = styled(Text)`
+  color: ${({ theme }) => theme.colors.fg.tertiary};
+`;
+
+const onPageIcons = (
+  <>
+    <Icons.Checkmark className="check" /> <Icons.Cross className="remove" />
+  </>
+);
+function ProductList({
+  products = [],
+  onPageProducts = [],
+  onInsert,
+  onRemove,
+  isMenuFocused = true,
+}) {
+  const { handleListNav, currentRowsRef, setCurrentFocusIndex } =
+    useProductNavigation({ isMenuFocused, products });
+
+  useEffect(() => {
+    if (!isMenuFocused) {
+      setCurrentFocusIndex(0);
+    }
+  }, [isMenuFocused, setCurrentFocusIndex]);
+
+  return (
+    // eslint-disable-next-line jsx-a11y/no-noninteractive-element-interactions -- list handles arrow up and arrow down
+    <div role="list" onKeyDown={handleListNav}>
+      {products.map((product, index) => {
+        const imageSrc = product?.productImages[0]?.url || '';
+        const onPage = isOnPage(product, onPageProducts);
+        const handleClick =
+          onPage && onPage?.productId
```

Comment:
Nit, since you are using optional chaining here:

```suggestion
          onPage?.productId
```